rest/slurmrestapi.py

```diff
@@ -425,6 +425,60 @@ def convert_nodeset():
     return json.dumps(list(NodeSet(data['nodeset'].encode('ascii', 'ignore'))))
 
 
+from utils import compact_nodes, expand_nodes
+@app.route('/sinfo', methods=['POST', 'OPTIONS'])
+@crossdomain(origin=origins, methods=['POST'],
+             headers=['Accept', 'Content-Type', 'X-Requested-With'])
+@cache()
+def sinfo():
+
+    # Partition and node lists are required 
+    # to compute sinfo informations
+    partitions = pyslurm.partition().get()
+    nodes = pyslurm.node().get()
+
+    # Retreiving the state of each nodes
+    nodes_state = dict(
+        (node.lower(), attributes['state'].lower())
+        for node, attributes in nodes.iteritems()
+    )
+
+    # For all partitions, retrieving the states of each nodes
+    sinfo_data = {}
+    for name, attr in partitions.iteritems():
+
+        for node in expand_nodes(attr['nodes']):
+            key = (name, nodes_state[node])
+            if key not in sinfo_data.keys():
+                sinfo_data[key] = []
+            sinfo_data[key].append(node)
+
+    # Preparing the response
+    resp = []
+    for k, v in sinfo_data.iteritems():
+        name, state = k
+        partition = partitions[name]
+        avail = partition['state'].lower()
+        min_nodes = partition['min_nodes']
+        max_nodes = partition['max_nodes']
+        total_nodes = partition['total_nodes']
+        job_size = "{0}-{1}".format(min_nodes, max_nodes)
+        job_size = job_size.replace('UNLIMITED', 'infinite')
+        time_limit = partition['max_time_str'].replace('UNLIMITED', 'infinite')
+        resp.append({
+          'name': name,
+          'avail': avail,
+          'job_size': job_size,
+          'time_limit': time_limit,
+          'nodes': total_nodes,
+          'state': state,
+          'nodelist': compact_nodes(v)
+        })
+
+    # Jsonify can works on list, thus using json.dumps
```

Comment:
I guess you meant **_cannot_** work on list, right? Or I don't get the point. Please note Flask `jsonify()` not only concert a hash into a JSON string, it also sets HTTP headesr for properly handling JSON content.